synapse/storage/databases/main/events.py

```diff
@@ -1943,15 +1943,43 @@ def _handle_batch_event(self, txn: LoggingTransaction, event: EventBase):
 
         txn.execute(sql, (batch_id,))
 
-    def _handle_redaction(self, txn, redacted_event_id):
+    def _handle_redact_relations(
+        self, txn: LoggingTransaction, redacted_event_id: str
+    ) -> None:
         """Handles receiving a redaction and checking whether we need to remove
         any redacted relations from the database.
 
         Args:
             txn
-            redacted_event_id (str): The event that was redacted.
+            redacted_event_id: The event that was redacted.
         """
 
+        # Fetch current relations.
+        relates_to = self.db_pool.simple_select_one_onecol_txn(
```

Comment:
I've just spent 5 minutes wondering how this `relates_to` differs for the `relates_to` in the event cache invalidation processing. Maybe worth renaming this to `redacted_relates_to`?

Reply:
Right, so this is slightly different because the redaction event doesn't refer to `relates_to`, we want the `relates_to` from the event being redacted.

So you have:

```mermaid
flowchart LR
A[Redaction] -->|redacts| B[Redacted Event]
B -->|m.relates_to| C[Related Event]
```

And we need to clear the caches for "Related Event". 👍